concordium-contracts-common/src/types.rs

```diff
@@ -331,6 +331,23 @@ impl ops::RemAssign<u64> for Amount {
     fn rem_assign(&mut self, other: u64) { *self = *self % other; }
 }
 
+/// A reference to a smart contract module deployed on the chain.
+#[repr(transparent)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ModuleReference([u8; 32]);
+
+impl convert::AsRef<[u8; 32]> for ModuleReference {
+    fn as_ref(&self) -> &[u8; 32] { &self.0 }
+}
+
+impl convert::From<[u8; 32]> for ModuleReference {
+    fn from(bytes: [u8; 32]) -> Self { Self(bytes) }
+}
+
+impl convert::From<ModuleReference> for [u8; 32] {
+    fn from(module: ModuleReference) -> Self { module.0 }
```

Comment:
Add `#[inline]`

Reply:
I have added inlining for this and the other converts introduce by this PR